pepper-apis/src/main/resources/db-changes/schema/refactor-invitations.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet author="yufeng" id="20200508-refactor-invitations">
+        <dropNotNullConstraint tableName="invitation" columnName="user_id" columnDataType="bigint"/>
+        <dropNotNullConstraint tableName="invitation" columnName="contact_email" columnDataType="varchar(255)"/>
+        <dropUniqueConstraint tableName="invitation" constraintName="invitation_guid_uk"/>
+
+        <addUniqueConstraint tableName="invitation" columnNames="study_id, invitation_guid" constraintName="invitation_study_uk"/>
```

Comment:
Cool, so we can still have multiple invitations per user.  So we could recruit someone with an invite, and then age do an age up, and everything works :)

Reply:
Actually, the `numRows` checks are just paranoid and not related to guid uniqueness.

I loosened the uniqueness constraints to allow studies to generate their own invitation codes instead of random UUIDs the system generates. For example `testboston` wants to have their own nice 10-digit invitation codes. If other studies want to generate their own invitation codes, we might run into conflicts. Also, it kind of makes sense that invitations are study-specific things, thus scoped to a study.

Reply:
I feel I have addressed question about uniqueness. Resolving.